std/range/package.d

```diff
@@ -9594,6 +9594,324 @@ auto refRange(R)(R* range)
     foo(r);
 }
 
+/**
+Bitwise adapter over Integral type Ranges. Consumes the range elements bit by bit.
+*/
+struct Bitwise(R)
+    if (isInputRange!R && isIntegral!(ElementType!R))
+{
+    alias ER = ElementType!R;
+    alias UER = Unsigned!ER;
+
+    private R parent;
+    private UER mask;
+    private int maskPos;
+    private ER accumulator;
+
+    private enum UER bitsNum = ER.sizeof * 8;
+
+    this()(auto ref R range)
```

Comment:
The compiler should be eliding the copies anyway. You shouldn't need to worry about it.